pandas/core/groupby/generic.py

```diff
@@ -1813,10 +1813,17 @@ def groupby_series(obj, col=None):
             # Try to consolidate with normal wrapping functions
             from pandas.core.reshape.concat import concat
 
+            axis_number = obj._get_axis_number(self.axis)
+            if axis_number:
+                obj = obj.T
```

Comment:
Rather than transpose I think should just specify the iterator, so something like:

```
if axis_number == 0:
    iter_func = obj.items
else:
    iter_func = obj.iterrows
```

Then change subsequent line to 

```python
results = [groupby_series(content, label) for label, content in iter_func()]
```

Reply:
I'm trying that, thanks